src/cpp/src/cache_manager.hpp

```diff
@@ -46,6 +46,17 @@ class CacheManager {
         const bool is_gpu = m_device.find("GPU") != std::string::npos;
         m_block_size = is_gpu ? gpu_block_size : cpu_block_size;
 
+        if (is_gpu) {
+            auto can_get_context = [&]() {
+                try {
+                    auto remote_context = m_request.get_compiled_model().get_context();
+                } catch (const std::exception& ex) {
+                    return false;
+                }
+                return true;
+            };
+            OPENVINO_ASSERT(can_get_context(), "Can not get context");
```

Comment:
It seems that you just replaced `Not Implemented` error with `Can not get context`. If that's true, then what's the point?

Reply:
Just add a assert if it's Not Implemented to disable contituous batching, if that's true, the gpu device can allocate kv cache with context here: https://github.com/openvinotoolkit/openvino.genai/blob/e312b49e6a874a4401632734554998f25b1c5265/src/cpp/src/cache_manager.hpp#L159
 if we don't add this assert here, it will crash in https://github.com/openvinotoolkit/openvino.genai/blob/e312b49e6a874a4401632734554998f25b1c5265/src/cpp/src/cache_manager.hpp#L159
 
I simplified logic and remove the judgments, please help to review if it's more reasonable, thanks.

Reply:
The new implementation moved `m_request.get_compiled_model().get_context()` so it would fail earlier. But it's still a failure. Is that what you need and why?

Reply:
Yes, it's what I need, fail here will be caught here: https://github.com/openvinotoolkit/openvino.genai/blob/e312b49e6a874a4401632734554998f25b1c5265/src/cpp/src/llm_pipeline.cpp#L180,

if it fail in https://github.com/openvinotoolkit/openvino.genai/blob/e312b49e6a874a4401632734554998f25b1c5265/src/cpp/src/cache_manager.hpp#L159, it will crash because the pipeline implementation is CB already.